File: broccoli/ember_template_compiler.js
var fs = require('fs'),
    path = require('path');


module.exports = function(input, output) {


  var result = "(function() {\nvar Ember = { assert: function() {}, FEATURES: { isEnabled: function() {} } };\n"

  var content = fs.readFileSync(input, {encoding: 'UTF-8'});
  content = content.replace('import Ember from "ember-metal/core";','');
  content = content.replace('export default EmberHandlebars;','');

  result += content;
  result += "\nexports.precompile = EmberHandlebars.precompile;"
  result += "\nexports.EmberHandlebars = EmberHandlebars;"
  result += "\n})();"

  var directory = path.dirname(output);
  if (!fs.existsSync(directory)) {
    fs.mkdirSync(directory);
  }

  fs.writeFileSync(output, result);
};

File: rakep/ember_template_compiler.rb


class EmberTemplateCompiler
  class << self 
    def generate(inputPath, outputPath)

      file_content = File.read(inputPath)
      file_content.gsub!('import Ember from "ember-metal/core";', '')
      file_content.gsub!('export default EmberHandlebars;', '')

      out = "(function() {\nvar Ember = { assert: function() {}, FEATURES: { isEnabled: function() {} } };\n"
      out << file_content
      out << "\nexports.precompile = EmberHandlebars.precompile;"
      out << "\nexports.EmberHandlebars = EmberHandlebars;"
      out << "\n})();"

      File.open(outputPath, 'w') do |file| 
        file.write(out) 
      end

    end
  end
end
